core/server/master/src/main/java/alluxio/master/file/BatchTracker.java

```diff
@@ -0,0 +1,26 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.master.file;
+
+/**
+ * Used to receive batches from FileSystemMaster methods.
+ *
+ * @param <T> batch type
+ */
+public interface BatchTracker<T> {
```

Comment:
what if the interface was just a streaming interface, and one such implementation is the batched version? So, we create something like:
```
interface ResultStream<T> {
  void submit(T item);
}
```

Then, we have a new `BatchedResultStream(batchSize)` which has a `List<T>`, and when it gets full, it sends it? Then, the whole batching concept is isolated in the `BatchedResultStream`, and the users don't have to decide whether or not to submit a batch.